src/transformers/generation/utils.py

```diff
@@ -1681,9 +1689,14 @@ def generate(
                         "This model does not support `cache_implementation='static'`. Please check the following "
                         "issue: https://github.com/huggingface/transformers/issues/28981"
                     )
-                model_kwargs["past_key_values"] = self._get_cache(
-                    generation_config.cache_implementation, batch_size, generation_config.max_length
-                )
+                model_kwargs["past_key_values"] = self._get_cache(generation_config.cache_implementation, batch_size, generation_config.max_length)
+                if self.config.is_encoder_decoder:
+                    # manually set another cache for cross attention
+                    encoder_outputs = model_kwargs["encoder_outputs"][0]
+                    model_kwargs["past_key_values"] = (
+                        model_kwargs["past_key_values"],
+                        self._get_cache("one_shot", encoder_outputs.shape[0],  encoder_outputs.shape[1], '_cross_attn_cache')
+                    )
```

Comment:
I wonder if there is a better way to do this, because in the encoder-decoder scenario we need a tuple of two caches  here according to the current design, but this seems hardcode and easy to break